### PR TITLE
Added integration tests for TaskLauncherSink

### LINKED ISSUE #174: TaskLauncherSink needs test using LocalTaskLauncher
Currently the unit tests use a  Test TaskLauncher.  However we need to have tests that use the LocalTaskLauncher.

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #174

### CODE REVIEW

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		Resource stdoutResource = context.getResource("file://" + taskOutputDirectory + "/stdout.log");
+		Resource stderrResource = context.getResource("file://" + taskOutputDirectory + "/stderr.log");
+
+		assertTrue("stdout.log was not created", stdoutResource.exists());
+		assertTrue("stderr.log was not created", stderrResource.exists());
+
+		assertTrue("stdout was not populated with content", waitForFileToBePopulated(stdoutResource));
+		assertTrue("stdout did not contain expected content", verifyTaskWasLaunched(stdoutResource));
+		assertTrue("stderr was populated with content", stderrResource.contentLength() == 0);
+	}
```
Comment (mminella) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: For integration tests, why can't we test against the task repository?

Comment (cppwfs) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: sure...

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	private DataSource dataSource;
+
+	private Map<String, String> properties;
+
+	private TaskExplorer taskExplorer;
+
+	@Before
+	public void setup() {
+		properties = new HashMap<>();
+		properties.put("server.port", "0");
```
Comment (mminella) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: Why do we need this?

**Comment by mminella** [comment]
Minor polish applied and merged as 6b9bb9481fb9501a35e86716b74ed0f5af986ffb and 0aca771f5dc172915d151449a1618d9c336ad6be

**Pull request closed without merging**